### PR TITLE
Add support for Linux Lutris library WoW installation discovery

### PR DESCRIPTION
**Pull request by fultonm**
Hello, and thanks for this wonderful addon manager!

My apology in advance for coming out of no where with an implementation. This is just something that worked for me and I wanted to share it. I'm happy to improve and maintain this feature in the future as well if you like it.

This provides an implementation for Linux addon directory discovery process.

The most frictionless way for Linux gamers to install non-Steam games is with the Lutris Library: https://lutris.net/](https://lutris.net/) and quite often we will use this platform instead of installing manually with Wine, as they maintain various settings and configuration to make it "just work".

The Library has a known configuration file in the user's home directory under `~/.config/lutris/system.yml` and we can follow the location configured for `game_path` to find possible Warcraft directories. Lutris currently has 3 entries for installing WoW, but all of them are exactly the same Battle.net launchers as far as directory structure goes. This just checks each one and runs with the first valid WoW path it finds.

In a few areas outside of the `warcraft.service.linux.ts` it was necessary to add an if statement for `isLinux` because an absolute path starting with C:\ was the installation path, instead of using the blizzardAgentPath which contains the Lutris library directory prefix. So, there's some string hacking there.

I'm happy to improve this as I probably missed something related to following the architecture so please let me know your feedback!

### CODE REVIEW

**Review by linaori — COMMENTED** [review]
As I personally have never used Lutris I've asked a friend, and they've confirmed that the product.db was indeed in a path that would be found in this code:
```
/home/<username>/Games/world-of-warcraft/drive_c/ProgramData/Battle.net/Agent/product.db
```

**Comment by fultonm** [comment]
90% of the time that's going to be the case because the default installation path is `~/<user>/Games/`, but it is configurable as a Lutris preference in `system.yml` if they want to move their folder to an external hard drive or not clutter the home folder.
However, I just checked a default configuration from installing Lutris for the first time and `system.yml` does not exist at least until a different library location is selected. So I'll update this to default to `~/<user>/Games/` if `~/.config/lutris` exists but the system file doesn't exist, or `game_path` is not present within the file.

**Comment by fultonm** [comment]
Ok, this should work better.

File: wowup-electron/src/app/services/warcraft/warcraft-installation.service.ts [hunk 1]
```diff
@@ -347,6 +357,17 @@ export class WarcraftInstallationService {
     return path.join(location, clientFolderName, executableName);
   }
 
+  private getFullLutrisProductPath(location: string, agentPath: string, clientType: WowClientType): string {
```
Comment (jliddev) [inline wowup-electron/src/app/services/warcraft/warcraft-installation.service.ts:360]: See above linux comment.

Adding a platform specific step to "getInstalledProducts", after 'decodedProducts' is called might be a decent way to move this?
(IE: this._impl.resolveProducts(decodedProducts) or something)

File: wowup-electron/src/app/services/warcraft/warcraft-installation.service.ts [hunk 2]
```diff
@@ -319,7 +324,12 @@ export class WarcraftInstallationService {
 
     const label = await this._translateService.get(`COMMON.CLIENT_TYPES.${typeName.toUpperCase()}`).toPromise();
 
-    const newLocation = this.getFullProductPath(legacyLocation, clientType);
+    let newLocation: string;
+    if (this._electronService.isLinux) {
```
Comment (jliddev) [inline wowup-electron/src/app/services/warcraft/warcraft-installation.service.ts:328]: See above linux comment

**Review by jliddev — CHANGES_REQUESTED** [review]
Looks fine overall, just some minor code shuffling changes.

**Comment by fultonm** [comment]
Sounds good and I definitely agree with those points. I should be able to explore these later this week.

**Comment by fultonm** [comment]
Sorry for the delay, I was doing job interviews and playing SoM this week 🤭

I think the `resolveProducts` solution was a pretty smooth way to do that. Let me know how this looks to you.

**Comment by jliddev** [comment]
I just did a merge of dev into your branch, should be conflict free.

**Comment by jliddev** [comment]
Are we cancelling this? seemed like it was almost done?

**Comment by fultonm** [comment]
Sorry, I was confused about what I did with the rebase. I was working on cleaning up the commits so my PR didn't include unrelated changes (40 files changed), but I see this PR now correctly shows the right amount of changes.

**Review by jliddev — APPROVED** [review]

**Pull request merged**